draft-rescorla-tls-esni.md

```diff
@@ -440,23 +477,29 @@ A server operating in Shared Mode uses PaddedServerNameList.sni as
 if it were the "server_name" extension to finish the handshake. It
 SHOULD pad the Certificate message, via padding at the record layer,
 such that its length equals the size of the largest possible Certificate
-(message) covered by the same ESNI key.
+(message) covered by the same ESNI key. Moreover, the server MUST
+include the "encrypted_server_name" extension in EncryptedExtensions,
+and the value of this extension MUST match PaddedServerNameList.nonce.
 
 ## Split Mode Server Behavior {#backend-server-behavior}
 
-The backend Server ignores both the "encrypted_server_name" and the
-"server_name" (if any) and completes the handshake as usual. If in
-Shared Mode, the server will still know the true SNI, and can use it
-for certificate selection. In Split Mode, it may not know the true
-SNI and so will generally be configured to use a single certificate.
-{{communicating-sni}} describes a mechanism for communicating the
-true SNI to the backend server.
+In Split Mode, the backend server may not know the true SNI and thus will 
+generally be configured to use a single certificate. In this case, the backend 
+server ignores the "encrypted_server_name" and may ignore the "server_name" 
+(if any) and completes the handshake as usual. If the backend server
+knows the true SNI, e.g., via the mechanism described in {{communicating-sni}},
+the backend server MAY use it to complete the handshake.
```

Comment:
This text seems OBE. This mechanism won't work unless the backend has nonce, so you might as well give it SNI too.

Reply:
Good catch. I'll make this point to the preceding section.